packages/contracts/test/test-utils/voting.ts

```diff
@@ -57,3 +58,33 @@ export async function advanceAfterVoteEnd(endDate: number) {
   await advanceTimeTo(endDate);
   expect(await getTime()).to.be.greaterThanOrEqual(endDate);
 }
+
+export async function voteWithSigners(
+  votingContract: Contract,
+  proposalId: number,
+  signers: SignerWithAddress[],
+  signerIds: {
+    yes: number[];
+    no: number[];
+    abstain: number[];
+  }
+) {
+  let promises = signerIds.yes.map(i =>
+    votingContract.connect(signers[i]).vote(proposalId, VoteOption.Yes, false)
+  );
+
+  promises.push(
```

Comment:
I guess you wanted to append the array from the no loop the array of the yes loop. otherwise `Promise.all` doesn't work as you would expect.
use `concat` instead of `push`: `promises = promise.concat(....)`

Reply:
Fixed 53df3eb707971b0343ca6ede46b2205b13e4198d

Reply:
Thx, strange that it worked